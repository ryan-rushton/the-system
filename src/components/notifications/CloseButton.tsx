import { motion } from 'framer-motion';
import { FC } from 'react';
import styles from './CloseButton.module.scss';

const Path = ({ d }: { d: string }) => (
  <motion.path fill="transparent" strokeWidth="3" stroke="#fefefe" strokeLinecap="round" d={d} />
);

interface Props {
  /** A function to be called when the close icon is clicked. */
  onClose(): void;
}

/**
 * A button that creates a close icon for the notifications. Extracted to it's own component so we can
 * isolate the svg creation.
 */
const CloseButton: FC<Props> = ({ onClose }) => (
  <button data-testid="close-button" onClick={onClose} className={styles.close}>
    <svg width="12" height="12" viewBox="0 0 12 12">
      <Path d="M 1 1 L 11 11" />
      <Path d="M 1 11 L 11 1" />
    </svg>
  </button>
);

export default CloseButton;
