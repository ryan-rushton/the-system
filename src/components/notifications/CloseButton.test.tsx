import { cleanup, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import CloseButton from './CloseButton';

describe('NavMenu', () => {
  afterEach(cleanup);

  test('it renders', () => {
    render(<CloseButton onClose={jest.fn()} />);

    expect(screen.getByTestId('close-button')).toBeInTheDocument();
  });

  test('it matches snapshot', () => {
    render(<CloseButton onClose={jest.fn()} />);

    expect(screen.getByTestId('close-button')).toMatchSnapshot();
  });

  // Don't need an on Enter keydown test as the main element is a button
  test('it calls the onClose prop on click', async () => {
    const onClose = jest.fn();
    render(<CloseButton onClose={onClose} />);

    await userEvent.click(screen.getByTestId('close-button'));

    expect(onClose).toHaveBeenCalledTimes(1);
  });
});
