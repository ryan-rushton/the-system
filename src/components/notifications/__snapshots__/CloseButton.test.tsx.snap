// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavMenu > it matches snapshot 1`] = `
<button
  class="_close_241e48"
  data-testid="close-button"
>
  <svg
    height="12"
    viewBox="0 0 12 12"
    width="12"
  >
    <path
      d="M 1 1 L 11 11"
      fill="transparent"
      stroke="#fefefe"
      stroke-linecap="round"
      stroke-width="3"
    />
    <path
      d="M 1 11 L 11 1"
      fill="transparent"
      stroke="#fefefe"
      stroke-linecap="round"
      stroke-width="3"
    />
  </svg>
</button>
`;
